File: ruby/algo_ladder/largest_product/largest_product.rb
# Find the largest product of any two numbers within a given array.

# Input: [5, -2, 1, -9, -7, 2, 6]
# Output: 63 (-9 * -7)

def largest_product (input_array)
  index_one = 0
  largest = input_array[index_one] * input_array[index_one + 1]
  while index_one < input_array.length - 1
    if input_array[index_one] * input_array[index_one + 1] > largest
      largest = input_array[index_one] * input_array[index_one + 1]
    end
    index_one += 1
  end
  return largest
end

p largest_product([5, -2, 1, -9, -7, 2, 6])

File: javascript/algoLadder/largestProduct/largestProduct.js
// Find the largest product of any two numbers within a given array.

// Input: [5, -2, 1, -9, -7, 2, 6]
// Output: 63 (-9 * -7)

const largestProduct = function (array) {
  let result = array[0] * array[1]
  for (let i = 0; i < array.length - 1; i += 1) {
      if (array[i] * array[i + 1] > result) {
        result = array[i] * array[i + 1]
    }
  }
  return result
}

console.log(largestProduct([5, -2, 1, -9, -7, 2, 6]))